dffml/util/net.py

```diff
@@ -8,6 +8,23 @@
 from .os import chdir
 
 
+# Hash Helper Function
+def secure_hash(string, algorithm, encoding="yes") -> str:
+    if encoding == "yes":
+        string = string.encode("utf-8")
+    if algorithm == "sha256":
+        return hashlib.sha256(string).hexdigest()
+    elif algorithm == "sha384":
+        return hashlib.sha384(string).hexdigest()
+
+
+def insecure_hash(string, algorithm="md5", encoding="yes") -> str:
+    if encoding == "yes":
+        string = string.encode("utf-8")
+    if algorithm == "md5":
+        return hashlib.md5(string).hexdigest()
```

Comment:
We shouldn't have a case where we return None here, the caller wouldn't be expecting that. We should either `raise` an exception if algorithm is not md5, or we should always use md5, which I'd say is a safe assumption for now. A safe assumption since you've made algorithm a keyword argument which defaults to md5, if you're worried about forward compatibility, you really aren't gaining anything by adding this keyword argument now. One could always add it as a keyword argument later and add more functionality, so it probably is best for us to just always return the md5 hash

Reply:
insecure_hash now always uses md5 hash.